File: map_ts/src/Path/path.ts
import type { PathId } from "./path-id.js";
import type { EntityId } from "../Entity/entity-id.js";
import type { PathState } from "./path-state.js";
import type { PathDataState } from "./path-data-state.js";

export type Path<
  TPayload = unknown
> = {
  id: PathId;

  source: EntityId;
  target: EntityId;

  state: PathState;
  dataState: PathDataState;

  payload?: TPayload;
};

File: map_python/src/Path/path.py
from dataclasses import dataclass
from typing import Generic, TypeVar

from ..Entity.entity_id import EntityId
from .path_data_state import PathDataState
from .path_id import PathId
from .path_state import PathState

T = TypeVar("T")


@dataclass(slots=True)
class Path(Generic[T]):
    id: PathId
    source: EntityId
    target: EntityId
    state: PathState
    data_state: PathDataState
    payload: T | None = None
